opnode/l1/source.go

```diff
@@ -100,6 +101,22 @@ func (s Source) L1BlockRefByNumber(ctx context.Context, l1Num uint64) (eth.L1Blo
 	return s.l1BlockRefByNumber(ctx, new(big.Int).SetUint64(l1Num))
 }
 
+func (s Source) L1BlockRefByHash(ctx context.Context, hash common.Hash) (eth.L1BlockRef, error) {
+	header, err := s.client.HeaderByHash(ctx, hash)
+	if err != nil {
+		return eth.L1BlockRef{}, err
+	}
+	l1Num := header.Number.Uint64()
+	parentNum := l1Num
```

Comment:
Does this need the intermediate variable here? Also should it be `if parentNum == 0`?

Reply:
I'm pretty sure that this is correct. The intent is to saturate subtraction at 0 (i.e. parent block of genesis is not well formed and has 0 as the parent block number).

See https://github.com/ethereum-optimism/optimistic-specs/pull/285/files#r836779844